### ISSUE TITLE
[che-theia] - open http preview in separate browser tab

### ISSUE THREAD

**Issue by sparkoo** [eclipse-che/che#18002]
### Is your enhancement related to a problem? Please describe.

This issue https://github.com/eclipse/che/issues/17840 will introduce option to expose user applications on subdomains in single-host mode. As we typically don't have certificate for such subdomains, so we're exposing those ports on HTTP only. As a result of that, che-theia can't show the preview, because HTTP content can't be loaded from HTTPS page.

![20200930_190031_1920x2160_scrot](https://user-images.githubusercontent.com/1160903/94716552-4f242280-034f-11eb-934a-6dade343b030.png)

**Bonus task/question (is this for separate issue?):**
In case of endpoint with attribute `protocol: 'https'`, it's exposed on HTTPS, but without valid certificate (we don't have one for subdomains in single-host). In that case, the "Potential security issue" browser page is shown. However, you can't "accept the risk" from the preview directly (I guess some browser security feature). You have to again open the URL in separate tab, then you can click the "accept the risk" button, and only then you can load the page in che-theia preview again. Can we do anything here to improve the user experience? 
![20200930_184744_1920x2160_scrot](https://user-images.githubusercontent.com/1160903/94716861-cce82e00-034f-11eb-8d42-71cce412b94d.png)

### Describe the solution you'd like

Che-theia should open HTTP URLs in new browser tab, instead of in-editor preview. In theory, when che-theia is on HTTP as well, it will work, but I guess we don't support that anymore.

### Describe alternatives you've considered

n/a

### Additional context

https://github.com/eclipse/che/issues/17840

**Comment by azatsarynnyy** [comment]
> In theory, when che-theia is on HTTP as well, it will work, but I guess we don't support that anymore

I'm not sure if we still support HTTP, but anyway it's not recommended as Che Theia WebView Plugin API works in secure contexts only, https://github.com/eclipse-theia/theia/pull/6465#issuecomment-557494879.

**Comment by azatsarynnyy** [comment]
I see the following ways how Preview URL UX can be improved for single-host mode :arrow_down: 

The Preview URL feature is configurable in Che Theia through user preferences:
![image](https://user-images.githubusercontent.com/1636395/94780644-74f20b80-03d1-11eb-9fc4-206124e38973.png)

A user can choose in which mode an application's preview URL should be opened, by default (embedded/detached).
For a new Workspace, it asks a user to choose a preferred mode once a corresponding Task is running.

There's the detailed description of all the possible modes:
https://github.com/eclipse/che-theia/tree/master/plugins/task-plugin#theia---che-task-plug-in

But this preference is respected by [Task Plugin](https://github.com/eclipse/che-theia/tree/master/plugins/task-plugin) only. So that it works for the user applications that were run with Tasks only.
For the applications whose ports are autodetected by [Ports Plugin](https://github.com/eclipse/che-theia/tree/master/plugins/ports-plugin), the only variant ATM is to preview its URLs in an embedded mini-browser widget.

There are two areas where it can be improved for single-host mode:

1. Make [Ports Plugin](https://github.com/eclipse/che-theia/tree/master/plugins/ports-plugin) respect [Preview URL mode user's preference](https://github.com/eclipse/che-theia/tree/master/plugins/task-plugin#theia---che-task-plug-in). So, for the autodetected apps (ports), the preview URL mode can be configured as well.
2. Detect we're running in a single-host mode and set a preview URL mode as `alwaysGoTo`, by the default.

**Comment by azatsarynnyy** [comment]
I'm moving the issue to `area/plugins` as it depends on Task and Ports Plugins.

**Comment by azatsarynnyy** [comment]
ATM, there's a way to work it around by opening a preview URL in a separate browser's tab manually.
However, for non-experienced users, it might be not obvious how to do it. So, setting P1. Feel free to reprioritize it.

**Comment by vitaliy-guliy** [comment]
Is this issue relevant?
Today I have installed CRC and deployed Che in single host mode.
Played a lot with go and node workspaces which are on Get Started pahe.
I haven't found any sample, which starts HTTP! server.
Every preview works on HTTPS.
Could it be such, that HTTP is exposed as HTTPS?

**Comment by vitaliy-guliy** [comment]
Even when I installed `http-server` by hands, it works over HTTPS
![Screenshot from 2021-03-01 18-36-15](https://user-images.githubusercontent.com/1655894/109528213-213d3880-7abd-11eb-94be-e4b09a4bf0b9.png)

**Comment by vitaliy-guliy** [comment]
I'm closing it as non relevant. Feel free to open it if the problem is still present.
